tools/log-collector/log_collector.go

```diff
@@ -474,258 +321,186 @@ func (l *LogCollector) zipDir() error {
 	return nil
 }
 
-// TODO()
-func (l *LogCollector) getResourceObjectsWithLabel(resourcePath string,
-	resource *apiv1.APIResource) (objects unstructured.UnstructuredList) {
-	allObjects := l.getResourceObjects(resourcePath, resource)
-	for _, object := range allObjects.Items {
-		objectLabel := object.GetLabels()
-		if len(objectLabel) != 0 {
-			if checkLabelExist(objectLabel, K8STrilioVaultLabel) {
-				objects.Items = append(objects.Items, object)
+// filterResourceObjects filter objects on the basis of some conditions
+func (l *LogCollector) filterResourceObjects(resourcePath string,
+	resource *apiv1.APIResource) (objects unstructured.UnstructuredList, err error) {
+
+	var allObjects unstructured.UnstructuredList
+
+	if (!resource.Namespaced && clusteredResources.Has(resource.Kind)) ||
+		(resource.Namespaced && !excludeResources.Has(resource.Kind)) {
+		log.Infof("Fetching '%s' Resource", resource.Kind)
+		allObjects = l.getResourceObjects(resourcePath, resource)
+
+		if resource.Name == CRD {
+			allObjects, err = filterCRD(allObjects)
+			if err != nil {
+				return objects, err
 			}
 		}
+
+		if resource.Name == Namespaces && !l.Clustered {
+			allObjects = filterNS(allObjects, l.Namespaces)
```

Comment:
rename func to `filterInputNS` or something better.. should be able to understand what is filterNS

Reply:
resolved